soup/rule_count.c

```diff
@@ -243,7 +243,7 @@ RuleCount
 rule_count(FILE *fp_in)
 {
 	size_t wordi = 0;
-	char word[WORD_BUFFER_SIZE+1];
```

Comment:
Not sure that the removal of this paranoia byte is a good idea, but we are willing to give it a try

Reply:
The paranoia byte doesn't help you since the test is against the size of the word buffer, ie. `sizeof (word) <= wordi` not the macro constant.  The byte would've help if `WORD_BUFFER_SIZE <= wordi` had been the expression.